### ISSUE TITLE
ST - V2 - Uploading a new Image to the gallery

### ISSUE THREAD

**Issue by jessprice** [FlyoverWorks/storytime#100]
When I upload a new image it populates in the bottom row under the 3rd, 4th or last image pushing the image that was there up. I think how it worked where it would be the new first image in the first row made sense.

I attached some screen shots for reference

![screen shot 2015-03-03 at 9 47 55 am](https://cloud.githubusercontent.com/assets/8793987/6466574/4de1ad42-c18e-11e4-8345-d514039286ca.png)
![screen shot 2015-03-03 at 10 08 35 am](https://cloud.githubusercontent.com/assets/8793987/6466575/4df541f4-c18e-11e4-83e1-6110446e1f29.png)
![screen shot 2015-03-03 at 10 13 37 am](https://cloud.githubusercontent.com/assets/8793987/6466576/4df63316-c18e-11e4-9a54-2ec636202f7d.png)

**Comment by dvanderbeek** [comment]
This is going to be complicated because of pagination and the way the rows are set up. I'm going to leave it as is for now and address this later.

**Comment by jessprice** [comment]
OK, its just strange and hard to follow where your image just went.. It used to just be the first image.

**Comment by dvanderbeek** [comment]
yeah, it’s not ideal. but i needed to add .row divs to keep everything lined up. adding the new image will require figuring out if the last row is full or not, moving the last image in each row to the next, and figuring out what to do with the last image if the last row was already full.


—

**Comment by eanlain** [comment]
Thoughts on reloading the media gallery partial instead of dealing with new images using JS? Slash, check out the fix above...

Also noticed some issues when uploading a few images in a row. After a few images, the 'done' callback would stop being triggered on successful uploads. I found a [stackoverflow post](http://missioncriticallabs.com/blog/lessons-learned-from-jquery-file-upload) on the issue which [referenced an issue with setting the dataType](http://missioncriticallabs.com/blog/lessons-learned-from-jquery-file-upload). Removing the dataType appears to have fixed the issue, triggering the 'done' callback after each upload.

**Comment by bcroesch** [comment]
Fix looks good to me. I've noticed issues with this in the past & it's nice to get rid of some of that conditional JS.

**Comment by eanlain** [comment]
Alrighty, fix merged into dev branch - v2.0.0.rc1.
